docker pull <invalid image> errors out with traceback

Invalid image name still errors out with traceback after #441's resolution.

> [vagrant@centos7-adb atomicapp]$ sudo atomicapp --verbose  run foo --provider docker
2016-02-22 08:57:56,036 - [INFO] - main.py - Action/Mode Selected is: run
2016-02-22 08:57:56,036 - [DEBUG] - main.py - Final parsed cmdline: run --verbose foo --provider docker
2016-02-22 08:57:56,036 - [DEBUG] - main.py - NuleculeManager init app_path: /var/lib/atomicapp/foo-4714d136e9ec
2016-02-22 08:57:56,036 - [DEBUG] - main.py - NuleculeManager init    image: foo
2016-02-22 08:57:56,037 - [DEBUG] - main.py - Request to unpack to foo to /var/lib/atomicapp/foo-4714d136e9ec
2016-02-22 08:57:56,037 - [INFO] - base.py - Unpacking image: foo to /var/lib/atomicapp/foo-4714d136e9ec
2016-02-22 08:57:56,159 - [INFO] - container.py - Pulling Docker image: foo
2016-02-22 08:57:56,159 - [DEBUG] - container.py - /usr/bin/docker pull foo
Using default tag: latest
Trying to pull repository docker.io/library/foo ... not found
Error: image library/foo:latest not found
2016-02-22 08:58:01,405 - [ERROR] - main.py - Could not pull Docker image foo
Traceback (most recent call last):
  File "/home/vagrant/atomicapp/atomicapp/cli/main.py", line 101, in cli_run
    nm.run(**argdict)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/main.py", line 224, in run
    self.nulecule = self.unpack(dryrun=dryrun, config=self.answers)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/main.py", line 129, in unpack
    nodeps=nodeps, dryrun=dryrun, update=update)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/base.py", line 104, in unpack
    docker_handler.pull(image)
  File "/home/vagrant/atomicapp/atomicapp/nulecule/container.py", line 63, in pull
    raise Exception("Could not pull Docker image %s" % image)
Exception: Could not pull Docker image foo

--- a/atomicapp/cli/main.py
+++ b/atomicapp/cli/main.py
@@ -37,7 +37,7 @@ from atomicapp.constants import (__ATOMICAPPVERSION__,
                                  LOGGER_DEFAULT,
                                  PROVIDERS)
 from atomicapp.nulecule import NuleculeManager
-from atomicapp.nulecule.exceptions import NuleculeException
+from atomicapp.nulecule.exceptions import NuleculeException, DockerException
 from atomicapp.utils import Utils
 
 logger = logging.getLogger(LOGGER_DEFAULT)
@@ -106,6 +106,9 @@ def cli_run(args):
         else:
             print_app_location(nm.app_path)
         sys.exit(0)
+    except DockerException as e:
+        logger.error(e)
+        sys.exit(1)
     except NuleculeException as e:
         logger.error(e)
         sys.exit(1)

--- a/atomicapp/nulecule/container.py
+++ b/atomicapp/nulecule/container.py
@@ -8,7 +8,7 @@ from atomicapp.constants import (APP_ENT_PATH,
                                  LOGGER_DEFAULT,
                                  MAIN_FILE)
 from atomicapp.utils import Utils
-from atomicapp.nulecule.exceptions import NuleculeException
+from atomicapp.nulecule.exceptions import NuleculeException, DockerException
 
 cockpit_logger = logging.getLogger(LOGGER_COCKPIT)
 logger = logging.getLogger(LOGGER_DEFAULT)
@@ -60,7 +60,7 @@ class DockerHandler(object):
         if self.dryrun:
             logger.info("DRY-RUN: %s", pull_cmd)
         elif subprocess.call(pull_cmd) != 0:
-            raise Exception("Could not pull Docker image %s" % image)
+            raise DockerException("Could not pull Docker image %s" % image)
 
         cockpit_logger.info('Skipping pulling Docker image: %s' % image)
 

--- a/atomicapp/nulecule/exceptions.py
+++ b/atomicapp/nulecule/exceptions.py
@@ -1,5 +1,9 @@
 # -*- coding: utf-8 -*-
 
 
+class DockerException(Exception):
+    pass
+
+
 class NuleculeException(Exception):
     pass

